### DISCUSSION TITLE
accessing go struct from plain javascript

### DISCUSSION THREAD

**Discussion by palacinka** [wailsapp/wails#1945] [Q&A] [answered]
Hello, sorry for newbie question. I am using template  `plain`  (`wails init -n wails-test-01 -t plain`)  how can I access returned go struct from plain javascript?  When I try to examine returned value in dev tools  it contains this extremely nested object that seems not to contain my data.

<img src="https://i.imgur.com/TborfFt.png">

app.go
```
type multiAnswer struct {
	name string   `json:"name"`
	age  int8     `json:"age"`
	urls []string `json:"urls"`
	//time int16
	time string `json:"time"`
}

func (a *App) SendMulti(name string) multiAnswer {

	return multiAnswer{
		name: "test",
		age:  50,
		urls: []string{"a", "b", "c"},
		time: time.UnixDate,
	}

}
```

main.js

```
window.sendmulti = function(){

    try {
        window.go.main.App.SendMulti("test-string-1")
            .then((result) => {

                 window.testvar = result
                //what to do with `result` here so I can access all fields?
   


            })
            .catch((err) => {
                console.error(err);
            });
    } catch (err) {
        console.error(err);
    }
}
```

**Comment by palacinka** [comment, marked as answer]
Looks like my only problem was lower case names of struct fields.  After capitalizing them it works. 😳

<img src="https://i.imgur.com/DXwIHp9.png">

app.go
```
type multiAnswer struct {
	Name string   `json:"name"`
	Age  int8     `json:"age"`
	Urls []string `json:"urls"`
	//time int16
	Time string `json:"time"`
}

func (a *App) SendMulti(name string) multiAnswer {

	return multiAnswer{
		Name: "test",
		Age:  50,
		Urls: []string{"a", "b", "c"},
		Time: time.UnixDate,
	}

}
```

main.js

```
window.sendmulti = function(){

    try {
        window.go.main.App.SendMulti("test-string-1")
            .then((result) => {
                document.getElementById("response_x").innerText = result['age'];
                document.getElementById("response_y").innerText = result['name'];
                window.testvar = result;



            })
            .catch((err) => {
                console.error(err);
            });
    } catch (err) {
        console.error(err);
    }



}
```

**Reply by leaanthony** [reply]
Yes. In Go, any struct field that is lowercase is private. It's also mentioned in the [struct binding docs](https://wails.io/docs/howdoesitwork#method-binding). Do you think this could be made clearer? We're always looking to improve docs.

**Comment by palacinka** [comment]
Personally I always look at examples first. If documentation contained one example where  struct is being passed to JavaScript (I found only examples with  string  or []string ) it would made it completely clear to me and saved me some time 😎
